test/e2e/set/scylladbmonitoring/scylladbmonitoring.go

```diff
@@ -436,3 +497,357 @@ func decodeGrafanaDashboardFromGZBase64String(s string) (*grafanaDashboard, erro
+	prometheusName := prometheusNameForScyllaDBMonitoring(sm)
+	svc, err := f.KubeAdminClient().CoreV1().Services(f.Namespace()).Create(ctx, &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusName,
+			Namespace: f.Namespace(),
+		},
+		Spec: corev1.ServiceSpec{
+			ClusterIP: corev1.ClusterIPNone,
+			Selector: map[string]string{
+				"app.kubernetes.io/managed-by": "prometheus-operator",
+				"app.kubernetes.io/name":       "prometheus",
+				"app.kubernetes.io/instance":   prometheusName,
+				"operator.prometheus.io/name":  prometheusName,
+				"prometheus":                   prometheusName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "web",
+					Port:     9090,
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: f.FieldManager(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return svc
+}
+
+func createExternalPrometheusInstanceWithoutTLS(ctx context.Context, f *framework.Framework, sm *scyllav1alpha1.ScyllaDBMonitoring, sa *corev1.ServiceAccount, svc *corev1.Service) *monitoringv1.Prometheus {
+	g.GinkgoHelper()
+
+	framework.By("Creating a Prometheus instance")
+	prom, err := f.PrometheusOperatorAdminClient().MonitoringV1().Prometheuses(f.Namespace()).Create(ctx, &monitoringv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusNameForScyllaDBMonitoring(sm),
+			Namespace: f.Namespace(),
+		},
+		Spec: monitoringv1.PrometheusSpec{
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				ServiceAccountName: sa.Name,
+				ServiceName:        pointer.Ptr(svc.Name),
+				SecurityContext: &corev1.PodSecurityContext{
+					RunAsNonRoot: pointer.Ptr(true),
+					RunAsUser:    pointer.Ptr[int64](65534),
+					FSGroup:      pointer.Ptr[int64](65534),
+				},
+				ServiceMonitorSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"scylla-operator.scylladb.com/scylladbmonitoring-name": sm.Name,
+					},
+				},
+				Web: &monitoringv1.PrometheusWebSpec{
+					PageTitle: pointer.Ptr("ScyllaDB Prometheus"),
+				},
+			},
+			Alerting: &monitoringv1.AlertingSpec{
+				Alertmanagers: []monitoringv1.AlertmanagerEndpoints{
+					{
+						Name: "scylla-monitoring",
+						Port: apimachineryutilintstr.FromString("web"),
+					},
+				},
+			},
+			RuleSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"scylla-operator.scylladb.com/scylladbmonitoring-name": sm.Name,
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: f.FieldManager(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return prom
+}
+
+func createExternalPrometheusInstanceWithTLS(ctx context.Context, f *framework.Framework, sm *scyllav1alpha1.ScyllaDBMonitoring, sa *corev1.ServiceAccount, svc *corev1.Service) *monitoringv1.Prometheus {
+	g.GinkgoHelper()
+
+	framework.By("Creating a Prometheus instance with TLS")
+	tlsSecretName := prometheusTLSSecretNameForScyllaDBMonitoring(sm)
```

Comment:
Again, lost a bit of context. Who creates this? Are we wiring the external prometheus instance with certs generated by the operator?